### DISCUSSION TITLE
What is the difference between an API and a webhook?

### DISCUSSION THREAD

**Discussion by Keko787** [community/community#140742] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

The people need to know! What is the difference between an API and a webhook and can you define them plz!!!

**Comment by PauloDrefahl** [comment, marked as answer]
# API 
An API (Application Programming Interface) is a set of rules and protocols that allow one software system to communicate with another. With APIs, you need to actively send a request to the server, and then the server responds with the requested data.

# Webhook
A webhook, on the other hand, is a way for one system to send real-time data to another when an event happens. Unlike APIs, where you pull data by making requests, webhooks push data to a URL you provide. When a certain event occurs (like a payment being made or a file being uploaded), the webhook sends data to your URL automatically.

### For Example:
- API: You make a request, and you get a response (pull data).
- Webhook: The server sends you data when an event occurs (push data).**
